src/HTTPResponse.hpp

```diff
@@ -20,11 +20,18 @@ class HTTPResponse
 	int getStatusCode() const;
 	const std::string &getStatusMessage() const;
 	void setStatusCode(int statusCode);
+	std::string getBody() const;
+	std::string getHeader(const std::string &name) const;
 
 	void setHeader(const std::string &name, const std::string &value);
 	void setBody(const std::string &body);
 	std::string toString() const;
 
+	bool isCGI() const;
```

Comment:
A response should be just a response. Just an HTTPResponse. The responseHandler could be eventually the instance interested in knowing if a response it is serving is a dynamic one or not. But an HTTPResponse object should not carry this information. Or how can you justify it?

Reply:
> A response should be just a response

Yes and being CGI is precisely part of being a response...

Being CGI, or not being CGI, is quintessential for a response, it is the essence of the response (not kidding) so I think it belongs here yes

Reply:
All right! I guess your lobby and me has different views about HTTPResponse and HTTPRequest objects. As far as it is an 'aware' choice I'm fine with it!

Being CGI or not being CGI is quintessential not for being a response but for become the right response for a certain request. The response that reaches the client will have no trace of its being generated through CGI or through a static page. No one will know and the HTTPResponse will also not know it. I can give you two equal in all their quintessential properties http responses (sent by a server): one generated through CGI and one generated through a static file, and you will not be able to tell the difference between them.

Reply:
I see your point and I think it is a really good point, it reminds me of environment variables being called meta-variables only inside the CGI (which I don't like)